code/modules/sanity/sanity_mob.dm

```diff
@@ -95,14 +95,21 @@
 	if(sanity_invulnerability)
 		return
 	var/vig = owner.stats.getStat(STAT_VIG)
+	var/sanity_damage_view = 0
 	for(var/atom/A in view(owner.client ? owner.client : owner))
+		if(ishuman(A))
+			var/mob/living/carbon/human/H = A
+			if(H.stat != DEAD && !H.internal && H.stats.getPerk(PERK_TOXIC_REVENGER) && !owner.internal && !owner.stats.getPerk(PERK_TOXIC_REVENGER))
```

Comment:
Seems like a griff perk

Reply:
Yes. Nothing bad about it, more social interactions.

What we should add there, is add cap and don't add toxin damage it it's already 30, and provide messages for affected that someone in the room smells like explosion on chemical factory.

Reply:
And maybe inform the player in some way, rather than it being silent but deadly